### ISSUE TITLE
"expects no type arguments, but 1 given" for generic class on pyi

### ISSUE THREAD

**Issue by cadl** [python/mypy#8106]
I am trying to define a generic class on the pyi, but the mypy always throw an `error: "xxx" expects no type arguments, but 1 given` error.

**code:**
```
$ tree
.
├── bar.py
├── bar.pyi
├── foo.py
└── foo.pyi
```

foo.py
``` python
class Query(object):
    pass
```

foo.pyi
``` python
from typing import Generic, TypeVar

T = TypeVar("T")


class Query(Generic[T]): ...
```

bar.py
``` python

from foo import Query


class FooBar(object):
    pass


def foo(l: Query[FooBar]):
    pass
```

**the mypy command and result:**

```
$ mypy --show-column-numbers foo.py bar.py
bar.py:8:12: error: "Query" expects no type arguments, but 1 given
Found 1 error in 1 file (checked 2 source files)
```

--------------

If I copy the type hint to `foo.py`, the error will disappear

new foo.py
```python
from typing import Generic, TypeVar

T = TypeVar("T")


class Query(Generic[T]):
    pass
```

```
$ mypy --show-column-numbers foo.py bar.py                                    
Success: no issues found in 2 source files
```
